src/gui/uithememanager.cpp

```diff
@@ -140,3 +154,81 @@ QString UIThemeManager::getIconPath(const QString &iconId) const
 #endif
     return IconProvider::instance()->getIconPath(iconId);
 }
+
+void UIThemeManager::loadColorsFromJSONConfig()
+{
+    const QString configFileName(":uitheme/config.json");
+    QFile configFile(configFileName);
+    if (!configFile.open(QIODevice::ReadOnly)) {
+        LogMsg(tr("Failed to open \"%1\". Reason: %2").arg(configFileName, configFile.errorString()), Log::WARNING);
+        return;
+    }
+
+    QJsonParseError jsonError;
+    const QJsonDocument configJsonDoc = QJsonDocument::fromJson(configFile.readAll(), &jsonError);
+    if (jsonError.error != QJsonParseError::NoError) {
+        LogMsg(tr("Invalid JSON provided in \"%1\". Reason: %2").arg(configFileName, jsonError.errorString()), Log::WARNING);
+        return;
+    }
+    if (!configJsonDoc.isObject()) {
+        LogMsg(tr("Invalid JSON provided in \"%1\". Reason: %2").arg(configFileName, tr("Root JSON value is not an object")), Log::WARNING);
+        return;
+    }
+
+    const QJsonObject colors = configJsonDoc.object().value("colors").toObject();
+    for (auto color = colors.constBegin(); color != colors.constEnd(); ++color) {
+        QColor providedColor(color.value().toString());
+        if (!providedColor.isValid()) {
+            LogMsg(tr("Invalid color for ID \"%1\" is provided by theme").arg(color.key()));
+            continue;
+        }
+        m_colors.insert(color.key(), providedColor);
+    }
+}
+
+void UIThemeManager::loadCustomPalette()
+{
+    struct ColorDescriptor
+    {
+        QString id;
+        QPalette::ColorRole colorRole;
+        QPalette::ColorGroup colorGroup;
+    };
+
+    const ColorDescriptor paletteColorDescriptors[] =
+    {
+        {"Palette.Window", QPalette::Window, QPalette::Normal},
```

Comment:
please wrap the first parameter with `QLatin1String` or even `QStringLiteral`.

Reply:
I wouldn't use QStringLiteral here. These strings are used only once. I don't want to increase binary size due to them.

Reply:
need to wrap around `QLatin1String` or should I leave it?

Reply:
>need to wrap around QLatin1String?

Yes.